### PR TITLE
[breaking change] Do not use Mustache but SpringBoot like references to refer to secrets and globals

### PR DESCRIPTION
**Pull request by eolivelli**
Problems statement:
1. We using Mustache both for resolving variables (secrets and globals) and for templating of Prompts and Queries.
This nesting of Mustache templates is really awkward and the use must use a strange '{{%' syntax in Prompts and Queries.
2. Mustache only produces strings, and it cannot return maps or lists, so you cannot use global variables for lists or maps
3. you must use the triple curly braces in order to not escape the values ({{{secrets.password}}} works, {{secrets.password}} sometimes doesn't work)

Modifications:
- Drop Mustache from the preprocessor, let's use a StringBoot like syntax ${secrets.x.y.z} and ${globals.x.y.z}
- Change all the examples to use the new syntax

### CODE REVIEW

File: langstream-core/src/main/java/ai/langstream/impl/common/ApplicationPlaceholderResolver.java [hunk 1]
```diff
@@ -47,6 +47,9 @@ public class ApplicationPlaceholderResolver {
     private static final ObjectMapper mapper =
             new ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);
 
+    private static final ObjectMapper mapperFprTemplates =
```
Comment (cbornet) [inline langstream-core/src/main/java/ai/langstream/impl/common/ApplicationPlaceholderResolver.java:50]:
```suggestion
    private static final ObjectMapper mapperForTemplates =
```

File: langstream-core/src/main/java/ai/langstream/impl/common/ApplicationPlaceholderResolver.java [hunk 2]
```diff
@@ -239,33 +242,82 @@ static Object resolveValue(Map<String, Object> context, Object object) {
...
+            if (end < 0) {
+                throw new IllegalArgumentException("Invalid placeholder: " + template);
+            }
+            String placeholder = template.substring(pos + 2, end);
+            Object value = resolveValue(context, placeholder);
+            if (!(value instanceof String)) {
+                // stuff that is not a String has to be converted to something that fits in a String
+                // using JSON is the least bad option
+                try {
+                    value = mapperFprTemplates.writeValueAsString(value);
```
Comment (cbornet) [inline langstream-core/src/main/java/ai/langstream/impl/common/ApplicationPlaceholderResolver.java:287]:
```suggestion
                    value = mapperForTemplates.writeValueAsString(value);
```

Comment (eolivelli) [reply langstream-core/src/main/java/ai/langstream/impl/common/ApplicationPlaceholderResolver.java]: committed, thanks

File: langstream-k8s-deployer/langstream-k8s-deployer-operator/src/test/java/ai/langstream/deployer/k8s/controllers/AppControllerIT.java [hunk 3]
```diff
@@ -400,7 +400,7 @@ private ApplicationCustomResource createAppWithResources(
                                 metadata:
                                   name: %s
                                 spec:
-                                    application: '{"agentRunners": {"agent1": {"resources": {"size": %d, "parallelism": %d}}}}'
+                                    application: '{"agentRunners": {"agent1": {"resources": {"size": %d, "parallelism": %d}}'
```
Comment (nicoloboschi) [inline langstream-k8s-deployer/langstream-k8s-deployer-operator/src/test/java/ai/langstream/deployer/k8s/controllers/AppControllerIT.java:403]: this is unrelated

File: langstream-k8s-deployer/langstream-k8s-deployer-core/src/test/java/ai/langstream/deployer/k8s/limits/ApplicationResourceLimitsCheckerTest.java [hunk 4]
```diff
@@ -124,7 +124,7 @@ private ApplicationCustomResource createApp(String tenant, int size, int paralle
                                   name: %s
                                   namespace: %s
                                 spec:
-                                    application: '{"agentRunners": {"agent1": {"resources": {"size": %d, "parallelism": %d}}}}'
+                                    application: '{"agentRunners": {"agent1": {"resources": {"size": %d, "parallelism": %d}}'
```
Comment (nicoloboschi) [inline langstream-k8s-deployer/langstream-k8s-deployer-core/src/test/java/ai/langstream/deployer/k8s/limits/ApplicationResourceLimitsCheckerTest.java:127]: unrelated

**Review by nicoloboschi — CHANGES_REQUESTED** [review]

File: langstream-k8s-deployer/langstream-k8s-deployer-core/src/test/java/ai/langstream/deployer/k8s/limits/ApplicationResourceLimitsCheckerTest.java (hunk 4 above)
Comment (eolivelli) [reply langstream-k8s-deployer/langstream-k8s-deployer-core/src/test/java/ai/langstream/deployer/k8s/limits/ApplicationResourceLimitsCheckerTest.java]: fixed

File: langstream-k8s-deployer/langstream-k8s-deployer-operator/src/test/java/ai/langstream/deployer/k8s/controllers/AppControllerIT.java (hunk 3 above)
Comment (eolivelli) [reply langstream-k8s-deployer/langstream-k8s-deployer-operator/src/test/java/ai/langstream/deployer/k8s/controllers/AppControllerIT.java]: fixed

**Review by nicoloboschi — APPROVED** [review]
